runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java

```diff
@@ -24,93 +25,549 @@
+	private TooltipManager tooltipManager;
+
+	@Inject
+	private MouseManager mouseManager;
 
-	private String getTags(int itemId)
+	@Inject
+	private BankTagsConfig config;
+
+	@Inject
+	private TabManager tabManager;
+
+	@Inject
+	private TagManager tagManager;
+
+	private int idx = 0;
+	private int maxTabs = 0;
+	private int preEditInt = 0;
+	private boolean scrollWait = false;
+	private boolean isBankOpen = false;
+	private Widget upButton = null;
+	private Widget downButton = null;
+	private Widget newTab = null;
+	private Widget parent = null;
+	private TagTab iconToSet = null;
+	private TagTab tagToRemove = null;
+	private TagTab activeTab = null;
+	private String preEditStr = "";
+	private Rectangle canvasBounds = new Rectangle();
+	private Rectangle bounds = new Rectangle();
+	private Map<Integer, SpritePixels> overrides = new HashMap<>();
+
+	@Provides
+	BankTagsConfig getConfig(ConfigManager configManager)
 	{
-		String config = configManager.getConfiguration(CONFIG_GROUP, ITEM_KEY_PREFIX + itemId);
-		if (config == null)
+		return configManager.getConfig(BankTagsConfig.class);
+	}
+
+	@Override
+	public void startUp()
+	{
+		parent = client.getWidget(WidgetInfo.BANK_CONTENT_CONTAINER);
+
+		if (parent != null)
 		{
-			return "";
+			loadTabs();
 		}
-		return config;
+
+		mouseManager.registerMouseWheelListener(this);
+
+		tagManager.convert();
+
+		Map<Integer, SpritePixels> overrides = new HashMap<>();
+		overrides.put(INCINERATOR, getSpritePixels(ImageUtil.getResourceStreamFromClass(getClass(), "incinerator.png")));
+		overrides.put(TAB_BACKGROUND, getSpritePixels(ImageUtil.getResourceStreamFromClass(getClass(), "tag-tab.png")));
+		overrides.put(TAB_BACKGROUND_ACTIVE, getSpritePixels(ImageUtil.getResourceStreamFromClass(getClass(), "tag-tab-active.png")));
+		overrides.put(UP_ARROW, getSpritePixels(ImageUtil.getResourceStreamFromClass(getClass(), "up-arrow.png")));
+		overrides.put(DOWN_ARROW, getSpritePixels(ImageUtil.getResourceStreamFromClass(getClass(), "down-arrow.png")));
+		overrides.put(NEW_TAG_TAB, getSpritePixels(ImageUtil.getResourceStreamFromClass(getClass(), "new-tab.png")));
+		client.setSpriteOverrides(overrides);
 	}
 
-	private void setTags(int itemId, String tags)
+	@Override
+	public void shutDown()
 	{
-		if (tags == null || tags.isEmpty())
+		if (upButton != null)
 		{
-			configManager.unsetConfiguration(CONFIG_GROUP, ITEM_KEY_PREFIX + itemId);
+			upButton.setHidden(true);
+			downButton.setHidden(true);
+			newTab.setHidden(true);
 		}
-		else
+
+		tabManager.clear();
+		tagManager.clear();
+		mouseManager.unregisterMouseWheelListener(this);
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged configChanged)
+	{
+		if (configChanged.getGroup().equals("banktags"))
+		{
+			if (configChanged.getKey().equals("rememberPosition"))
+			{
+				config.position(config.rememberPosition() ? idx : 0);
+			}
+			else if (configChanged.getKey().equals("useTabs"))
```

Comment:
Yeah, this entire if is wrong. Just remove the useTabs config as I mentioned before completely